### PR TITLE
SC/collectives: add change logs

### PR DESCRIPTION
**Pull request by minsii**

### CODE REVIEW

**Comment by minsii** [comment]
This PR contains several fixes and change logs for the collective section. @nspark @jdinan @davidozog Please review.

File: content/backmatter.tex [hunk 1]
```diff
@@ -522,6 +522,25 @@ \section{Version 1.5}
 \CONST{SHMEM\_MALLOC\_ATOMICS\_REMOTE} and \CONST{SHMEM\_MALLOC\_SIGNAL\_REMOTE}.
 \\ See Section \ref{subsec:shmmallochint} and \ref{subsec:library_constants}.
 %
+\item Changed the broadcast function to update the dest object on all PEs,
+including the root PE.
+\\ See Section \ref{subsec:shmem_broadcast}.
```
Comment (jdinan) [inline content/backmatter.tex:527]: This change is only for team-based broadcasts, not active-set based broadcasts. Can you please clarify?

File: content/backmatter.tex [hunk 2]
```diff
@@ -522,6 +522,25 @@ \section{Version 1.5}
...
+%
+\item Clarified error codes returned from collectives.
+\\ See Section \ref{subsec:coll}.
```
Comment (jdinan) [inline content/backmatter.tex:530]: I believe that the active-set based collectives did not return error codes. Since this only applies to the new collectives, it seems unnecessary to note this as a separate changelog entry.

File: content/backmatter.tex [hunk 3]
```diff
@@ -522,6 +522,25 @@ \section{Version 1.5}
...
+%
+\item Deprecated entire active-set-based collectives. Added the
+  team-based collective functions with return value to indicate successful
+  local completion: \FUNC{shmem\_sync},
```
Comment (jdinan) [inline content/backmatter.tex]: Can we split this up?
```
\item Deprecated active-set-based collectives.
\item Added team-based collective functions:
```
The return value doesn't seem to me like a detail that needs to be called out here?

File: content/shmem_sync_all.tex [hunk 4]
```diff
@@ -1,6 +1,6 @@
 \apisummary{
     Registers the arrival of a \ac{PE} at a synchronization point and suspends
-    execution until all other \acp{PE} in the default team arrive at a synchronization point. For multithreaded programs, execution is suspended
+    execution until all other \acp{PE} in the default team arrive at the synchronization point. For multithreaded programs, execution is suspended
```
Comment (jdinan) [inline content/shmem_sync_all.tex]: Can we replace "default team" with `SHMEM_TEAM_WORLD` or "world team"?

File: content/backmatter.tex (hunk 2 above)
Comment (minsii) [reply content/backmatter.tex:530]: Will delete the change log. However, the main text does not specify that the error codes are team-based collectives specific. How about we change the title?
```
Error codes returned from collectives
```
->
```
Error codes returned from team-based collectives
```

Comment (jdinan) [reply content/backmatter.tex:530]: Did we update the legacy collectives to also return error codes? I think this was discussed but ultimately not incorporated.

Comment (minsii) [reply content/backmatter.tex:530]: We didn't. Same for `shmem_barrier_all` and `shmem_sync_all`.  Do you think we should leave the paragraph title as is?

File: content/shmem_sync_all.tex (hunk 4 above)
Comment (minsii) [reply content/shmem_sync_all.tex]: "default team" is used in several sections including teams, collectives, and memory management. It is defined in section 7. Library Handles.

I agree that `SHMEM_TEAM_WORLD` is more specific. Perhaps we want to replace it everywhere for consistency. Do we need to create separate commits for each SC? Is there a way that we can fix it in a single commit?

Comment (nspark) [reply content/shmem_sync_all.tex]: I'd prefer "world team"—since, as with contexts, one could copy the handle to `SHMEM_TEAM_WORLD` and still refer to the same team—but this would be a "whole document" change (which I'd support).

File: content/collective_intro.tex [hunk 5]
```diff
@@ -55,7 +55,7 @@ \subsubsection*{Team-based collectives}
 \item \FUNC{shmem\_\{TYPE\_\}broadcast\{mem\}}
 \item \FUNC{shmem\_\{TYPE\_\}collect\{mem\}}
 \item \FUNC{shmem\_\{TYPE\_\}fcollect\{mem\}}
-\item Reduction routines for the following operations: AND, OR, XOR, MAX, MIN, SUM, PROD
+\item Reduction routines for the following operations: AND, MAX, MIN, SUM, PROD, OR, XOR
```
Comment (nspark) [inline content/collective_intro.tex]: Why the reordering? The new ordering isn't alphabetized, which would be: AND, MAX, MIN, OR, PROD, SUM, XOR. The old order was, as I saw it, ordered as {bitwise, comparison, arithmetic} operators.

File: content/shmem_broadcast.tex [hunk 6]
```diff
@@ -145,7 +145,7 @@
 
 \apicexample
     {In the following \Cstd[11] example, the call to \FUNC{shmem\_broadcast} copies \source{}
-    on \ac{PE} $0$ to \dest{} on \acp{PE} $1\dots npes-1$.
+    on \ac{PE} $0$ to \dest{} on \acp{PE} $0\dots npes-1$.
```
Comment (nspark) [inline content/shmem_broadcast.tex:147]: Good catch!

File: content/collective_intro.tex (hunk 5 above)
Comment (minsii) [reply content/collective_intro.tex]: I reordered it to match with the description for active-set based collectives. It looks weird to me that they describe very similar contents but in different orders. Maybe it is better to use "AND, OR, XOR, MAX, MIN, SUM, PROD" for both.  Suggestion?

Comment (nspark) [reply content/collective_intro.tex]: Yes, we should reorder the line for the active-set based collectives to match that of the team-based collectives:

https://github.com/nspark/specification/blob/14e66bed2fceda1567310f9e2cb4b5bf15056250/content/collective_intro.tex#L115

This will match the ordering of the routine listings in the Specification itself.

File: content/shmem_sync_all.tex (hunk 4 above)
Comment (minsii) [reply content/shmem_sync_all.tex]: Good point on "world team". I created issue #371 to track the changes. I'd deal with the updates in separate commits and submit to the corresponding SC.

Comment (jdinan) [reply content/shmem_sync_all.tex]: @nspark I'm fine with "world team" but we need to be precise about what that means. If one thread is calling `shmem_malloc`, is it a violation of the threading rules for another to call `shmem_team_barrier(SHMEM_TEAM_WORLD)` at the same time?

File: content/backmatter.tex (hunk 2 above)
Comment (jdinan) [reply content/backmatter.tex:530]: A clarification is something we would make to existing text. Given that the error codes are newly added in OpenSHMEM 1.5, this changelog entry seems confusing to me. I think it would be fine to remove it and assume that this change is covered by another entry indicating the new team-based collectives. However, if you feel like it's helpful to call out this change in particular, it could be phrased a bit more like a new addition:

> Specified that \openshmem defines only success and failure error codes for team-based collective routines.

File: content/collective_intro.tex (hunk 5 above)
Comment (minsii) [reply content/collective_intro.tex]: Fixed.

File: content/shmem_sync_all.tex (hunk 4 above)
Comment (minsii) [reply content/shmem_sync_all.tex]: Thinking again about the example @nspark mentioned: one does `handle = SHMEM_TEAM_WORLD;` and uses `handle` in later calls. The value of `handle` is still the predefined `SHMEM_TEAM_WORLD`, and the function call reads the value but not `&handle`.  Thus, I don't think people may confuse if the `handle` is a different team.

However, if we say "world team", I am afraid that people may confuse if a duplicated team of `SHMEM_TEAM_WORLD` (e.g., created by `shmem_team_split_strided`) is also a "world team".

In summary, I am now thinking `SHMEM_TEAM_WORLD` might be a better replacement for "default team." Does it resolve @jdinan's question, or avoid such confusion?

**Comment by jdinan** [comment]
You could also specify in the description of the SHMEM_TEAM_WORLD handle
that it corresponds to the "world team".

File: content/shmem_sync_all.tex (hunk 4 above)
Comment (jdinan) [reply content/shmem_sync_all.tex]: In Section 7, you can clarify that `SHMEM_TEAM_WORLD` handle corresponds to the "world team". Currently it says:
> Handle of type shmem_team_t that corresponds to the default team of all PEs in the OpenSHMEM program.

File: content/backmatter.tex (hunk 2 above)
Comment (minsii) [reply content/backmatter.tex:530]: Sorry for my late reply. I agree that the error code change is covered by the new team-based collectives entry. Thus, I deleted the change log entry and updated the paragraph title to: "Error codes returned from team-based collectives"

File: content/shmem_sync_all.tex (hunk 4 above)
Comment (minsii) [reply content/shmem_sync_all.tex]: I am working on the default team->world team replacement now. It will be in separate PRs.

**Comment by minsii** [comment]
@nspark @jdinan @davidozog I think all comments should be fixed except the "default team"->"world team" replacement. I will address it in separate PRs. This PR should be ready for review & merge. Could you please check again?

File: content/backmatter.tex [hunk 7]
```diff
@@ -522,6 +522,23 @@ \section{Version 1.5}
 \CONST{SHMEM\_MALLOC\_ATOMICS\_REMOTE} and \CONST{SHMEM\_MALLOC\_SIGNAL\_REMOTE}.
 \\ See Section \ref{subsec:shmmallochint} and \ref{subsec:library_constants}.
 %
+\item Changed the team-based broadcast function to update the dest object on
+all PEs, including the root PE.
```
Comment (nspark) [inline content/backmatter.tex]:
```suggestion
all \acp{PE}, including the root \ac{PE}.
```

File: content/backmatter.tex [hunk 8]
```diff
@@ -522,6 +522,23 @@ \section{Version 1.5}
 \CONST{SHMEM\_MALLOC\_ATOMICS\_REMOTE} and \CONST{SHMEM\_MALLOC\_SIGNAL\_REMOTE}.
 \\ See Section \ref{subsec:shmmallochint} and \ref{subsec:library_constants}.
 %
+\item Changed the team-based broadcast function to update the dest object on
```
Comment (nspark) [inline content/backmatter.tex]:
```suggestion
\item Changed the team-based broadcast function to update the \VAR{dest} object on
```

**Review by nspark — CHANGES_REQUESTED** [review]
Looks good; only requesting two minor formatting changes

File: content/backmatter.tex [hunk 9]
```diff
@@ -522,6 +522,23 @@ \section{Version 1.5}
 \CONST{SHMEM\_MALLOC\_ATOMICS\_REMOTE} and \CONST{SHMEM\_MALLOC\_SIGNAL\_REMOTE}.
 \\ See Section \ref{subsec:shmmallochint} and \ref{subsec:library_constants}.
 %
+\item Changed the team-based broadcast function to update the \VAR{dest} object on
```
Comment (nspark) [inline content/backmatter.tex]:
```suggestion
\item Specified that team-based broadcast operations update the \VAR{dest} object on
```

**Review by nspark — APPROVED** [review]
